### PR TITLE
implemented upload_geometry() function

### PR DESCRIPTION
**Pull request by NoaMills-USDA**
Accepts geojson, shp, csv, and zip files.

### CODE REVIEW

**Comment by HeatherSavoy-USDA** [comment]
Thanks! I have two small comments about method names:

1. I think when we were discussing rgeocdl and pygcdl function/method names, we did agree on underscores like you implemented, but I think we also said that "getDatasetInfo" sounded more informative than "view_metadata". So can we do "get_dataset_info"? (We can do another PR for rgeocdl to make it match.)
2. The getDatasets() and getDatasetInfo() in client_example.py should then be updated.

**Comment by BrianStucky-USDA** [comment]
I agree with both of Heather's comments! I have a few additional comments, too.

First some coding style suggestions:
* Your new code uses tabs for indentation, but we use spaces (4 spaces per indentation level) throughout GeoCDL. Please configure your editor to use spaces instead of tabs, as this is the standard community practice for Python code and also [recommended by](https://peps.python.org/pep-0008/#indentation) [PEP 8](https://peps.python.org/pep-0008/#tabs-or-spaces).
* Comments should include a space after the `#` and should usually be complete sentences, again, per [PEP 8](https://peps.python.org/pep-0008/#comments).
* I typically limit line lengths to 80 characters. This more or less follows [the PEP 8 recommendations](https://peps.python.org/pep-0008/#maximum-line-length) and generally makes code more readable (IMO).
* Using blank lines to separate logical sections of code can really help with readability, too.
* Functions should include a docstring that describes the function semantics and interface.
* `=` for a keyword argument in a function call should not have spaces next to it; again, per [PEP 8](https://peps.python.org/pep-0008/#pet-peeves).

And then four other comments:
* Creating a temporary directory for staging the ZIP file contents should use a `TemporaryDirectory` object from the [tempfile](https://docs.python.org/3/library/tempfile.html) module. `TemporaryDirectory` should be used as a context manager. This will ensure that:
  * The temporary directory is created in the proper, OS-specific location.
  * All files are cleaned up even if an exception occurs during processing.
* There are what appear to be debug `print()` statements in the code that should be removed.
* `shx_file_path` is created but the path is never validated.
* You might consider using the newer [pathlib](https://docs.python.org/3/library/pathlib.html) filesystem interface instead of the older `os.path` methods. The former provides better abstractions and more readable code (IMO).

**Comment by BrianStucky-USDA** [comment]
I just now pushed [a commit](86b3cadf51f53c25aee39fcce4516399a6ce000f) that gives an example of how I might format the code and comments you added. I hope that helps!

File: pygcdl.py [hunk 1]
```diff
@@ -174,11 +216,16 @@ def _format_geometry(self, endpoint, geom):
...
+
             if len(geom) == 36 and not "." in geom:
                 spatial_subset += "&geom_guid=" + geom
             else: #assume geom is a filename, attempt to upload
                 geom_guid = self.upload_geometry(geom)
                 spatial_subset += "&geom_guid=" + geom_guid
+        elif isinstance(geom, list) and \
+            all(len(geom[i]) == 36 for i in range(len(geom))):
+            for i in range(len(geom)):
+                spatial_subset += "&geom_guid" + geom[i]
```
Comment (HeatherSavoy-USDA) [inline pygcdl.py]: Is this concatenating multiple GUIDs together in one string?

Comment (HeatherSavoy-USDA) [inline pygcdl.py]: Missing = in "&geom_guid"

File: README.md [hunk 2]
```diff
@@ -1,5 +1,18 @@
-# pygcdl
-
-Python interface for the Geospatial Common Data Library.
-
-
+# pygcdl
+
+Python interface for the Geospatial Common Data Library.
+
+To run the script client_example.py, you must first unzip the sample datafiles, ie:
```
Comment (HeatherSavoy-USDA) [inline README.md:5]: The examples should be as simple to use as possible. If you want to stick to these counties, you could strip all (or most) attributes, transform the geometries to EPSG:4326 (if they aren't in the CRS already), and save the result as geojson here instead. Then they would each be a single file with no unzipping needed and less unnecessary data included.

**Review by HeatherSavoy-USDA — CHANGES_REQUESTED** [review]
I'm trying out this review feature for the first time! 🎉

**Comment by NoaMills-USDA** [comment]
Outdated PR

**Pull request closed without merging**